src/WireSlaveRequest.cpp

```diff
@@ -104,7 +104,7 @@ size_t WireSlaveRequest::available()
 int WireSlaveRequest::read()
 {
     int value = -1;
-    if (lastStatus_ == PACKET_READ && rxIndex_ < rxLength_) {
+    if (lastStatus_ == PACKET_READ && rxIndex_ < rxLength_ && (rxIndex_ < UNPACKER_BUFFER_LENGTH)) {
```

Comment:
Redundant check, can be how it was previously.

Reply:
```suggestion
    if (lastStatus_ == PACKET_READ && rxIndex_ < rxLength_) {
```